Docs locked params (#72657)
metabase/metabase: docs/embedding/guest-embedding.md

The end user won't see the "category" filter, but the dashboard will only show data for the "Gadget" category (or whatever you passed to the "category" array in the `params` object in the server payload you used to sign the JWT).

## Updating a locked parameter

Things to keep in mind if you need to make changes to your locked parameters.

### Include all locked parameters in your JWT

Once you publish a question or dashboard with a locked parameter, you _must_ include the name of the locked parameter in the `params` object when you sign the JWT. If you leave the parameter out, Metabase will refuse the request and log: `You must specify a value for :parameter in the JWT`.

> ### To turn off a locked parameter, pass an empty array as its value

Reviewer: Heading too long, I as a reader would ignore it

> ### To turn off a locked parameter, pass an empty array as its value

Author: ```suggestion
### Pass an empty array to turn off a locked parameter
```

If you don't want the locked filter to apply for a given token, pass an empty array, `[]`, as the value for the parameter in the JWT:

```javascript
const payload = {
  resource: { dashboard: 10 },
  params: {
    category: [], // locked filter is bypassed for this token
  },
  exp: Math.round(Date.now() / 1000) + 10 * 60,
};
```

This is handy when you want to reuse the same dashboard or question across contexts, and conditionally skip a locked filter in some of them.

> ### Make sure the filter name matches the locked parameter name

Reviewer: ```suggestion
### Filter name should match the locked parameter name
```

> If you rename a query builder filter that's used as a locked parameter, update the matching key in your JWT `params` object. Locked parameters that are connected to a [SQL variable](../questions/native-editor/sql-parameters.md) don't need to be renamed on the server.

Author: ```suggestion
If you rename a dashboard filter that's used as a locked parameter, update the matching key in your JWT `params` object. Locked parameters that are connected to a [SQL variable](../questions/native-editor/sql-parameters.md) don't need to be renamed on the server.
```

> If you rename a query builder filter that's used as a locked parameter, update the matching key in your JWT `params` object. Locked parameters that are connected to a [SQL variable](../questions/native-editor/sql-parameters.md) don't need to be renamed on the server.

Reviewer: What does rename a query builder filter mean? Query builder filters aren’t named (and also can't be used as locked params)

## Locked parameters limit the values available to other editable parameters

Because locked parameters filter data _before_ the results are displayed in the embed, locked parameters also limit the values available to any editable filter widgets on the same item.

For example, say you're embedding a dashboard with two filters, State and City. If you lock the State parameter with the value "Vermont", the City filter will only show cities in Vermont in its dropdown. You don't need to wire the filters together explicitly — they'll behave like [linked filters](../dashboards/filters.md#linking-filters).

## Locked parameters on dashboards with SQL questions

If a locked parameter is linked to a dashboard filter that's in turn linked to a SQL question, you'll only be able to pass a _single_ value for that locked parameter in the JWT.

For example, if you have a dashboard filter called "Breakfast" with the values "Hash browns", "Muffin", and "Waffles", and the filter is linked to _any_ SQL question on the dashboard, you can only pass one of those options as the locked parameter value.

> ## Using locked parameters to power custom widgets in your app

Reviewer: This could use an example. This section promises some cool stuff but doesn't make it clear how exactly i should accomplish that
